lib/blather/client/client.rb

```diff
@@ -277,7 +277,7 @@ def ready!
 
     def client_post_init
       write_with_handler Stanza::Iq::Roster.new do |node|
-        roster.process node
+        roster.process(node) unless node.error?
```

Comment:
I'm perfectly fine with the code simply not failing and the resulting state of an empty roster. 
I don't think that this edge case should be supported further.